src/gui/src/globalConnectDialog.cpp

```diff
@@ -0,0 +1,305 @@
+
+  adjustSize();
+}
+
+void GlobalConnectDialog::addRule(odb::dbGlobalConnect* gc)
+{
+  const int row_idx = rules_.size() + 1;
+  auto& widgets = rules_[gc];
+
+  widgets.inst_pattern = new QLineEdit(this);
+  widgets.inst_pattern->setReadOnly(true);
+  widgets.inst_pattern->setText(QString::fromStdString(gc->getInstPattern()));
+  layout_->addWidget(widgets.inst_pattern,
+                     row_idx,
+                     toValue(GlobalConnectField::Instance),
+                     Qt::AlignCenter);
+  widgets.pin_pattern = new QLineEdit(this);
+  widgets.pin_pattern->setReadOnly(true);
+  widgets.pin_pattern->setText(QString::fromStdString(gc->getPinPattern()));
+  layout_->addWidget(widgets.pin_pattern,
+                     row_idx,
+                     toValue(GlobalConnectField::Pin),
+                     Qt::AlignCenter);
+  widgets.net = new QLineEdit(this);
+  widgets.net->setReadOnly(true);
+  widgets.net->setText(QString::fromStdString(gc->getNet()->getName()));
+  layout_->addWidget(
+      widgets.net, row_idx, toValue(GlobalConnectField::Net), Qt::AlignCenter);
+
+  widgets.region = new QLineEdit(this);
+  odb::dbRegion* dbregion = gc->getRegion();
+  if (dbregion == nullptr) {
+    widgets.region->setText("All");
+  } else {
+    widgets.region->setText(QString::fromStdString(dbregion->getName()));
+  }
+  widgets.region->setReadOnly(true);
+  layout_->addWidget(widgets.region,
+                     row_idx,
+                     toValue(GlobalConnectField::Region),
+                     Qt::AlignCenter);
+
+  widgets.run = new QPushButton(this);
+  widgets.run->setIcon(QIcon(":/play.png"));
+  widgets.run->setToolTip(tr("Run"));
+  widgets.run->setAutoDefault(false);
+  widgets.run->setDefault(false);
+  connect(widgets.run, &QPushButton::pressed, this, [this, gc]() {
+    block_->globalConnect(gc);
+  });
+  layout_->addWidget(
+      widgets.run, row_idx, toValue(GlobalConnectField::Run), Qt::AlignCenter);
+
+  widgets.remove = new QPushButton(this);
+  widgets.remove->setIcon(QIcon(":/delete.png"));
+  widgets.remove->setToolTip(tr("Delete"));
+  widgets.remove->setAutoDefault(false);
+  widgets.remove->setDefault(false);
+  connect(widgets.remove, &QPushButton::pressed, this, [this, gc]() {
+    deleteRule(gc);
+  });
+  layout_->addWidget(widgets.remove,
+                     row_idx,
+                     toValue(GlobalConnectField::Remove),
+                     Qt::AlignCenter);
+
+  adjustSize();
+}
+
+void GlobalConnectDialog::makeRule()
+{
+  const std::string inst_pattern = inst_pattern_->text().toStdString();
+  const std::string pin_pattern = pin_pattern_->text().toStdString();
+  odb::dbNet* net = net_->currentData().value<odb::dbNet*>();
+  odb::dbRegion* region = region_->currentData().value<odb::dbRegion*>();
+
+  try {
+    auto* rule
+        = odb::dbGlobalConnect::create(net, region, inst_pattern, pin_pattern);
+    addRule(rule);
+  } catch (const std::runtime_error&) {
+  }
```

Comment:
why the try-catch with no handling?

Reply:
unlike the TCL interface, which wraps all commands with a try/catch, when the GUI calls into OpenROAD, it needs to catch the errors. The create can generate an exception (error) when the regex is invalid.

Reply:
Doesn't the user need some feedback in that case?

Reply:
The logger will have the error message, but I could generate a pop-up message box as well.

Reply:
It think so - I didn't notice the logger msg

Reply:
I added a popup box if an error occurs